packaging/Makefile

```diff
@@ -1,64 +1,60 @@
+.PHONY: docker.build.local
+docker.build.local: clean $(addprefix build/,$(BINARIES)) ## build local architecture image
+	docker build -t $(IMAGE) --build-arg TARGETPLATFORM= .
 
-routesrv:
-	GO111MODULE=$(GO111) GOOS=$(GOOS) GOARCH=$(GOARCH) $(GOARM) CGO_ENABLED=$(CGO_ENABLED) go build -o routesrv ../cmd/routesrv/*.go
+.PHONY: docker.build.amd64
+docker.build.amd64: clean build.linux.amd64 docker.build.enable ## build linux/amd64 image using a trusted amd64 alpine base image
+	docker buildx build -t $(IMAGE) --platform linux/amd64 --load . \
+		--build-arg BASE_IMAGE=registry.opensource.zalan.do/library/alpine-3:latest
 
-clean:
-	rm -rf $(BINARIES) build/
+.PHONY: docker.build.arm64
+docker.build.arm64: clean build.linux.arm64 docker.build.enable ## build linux/arm64 image
+	docker buildx build -t $(ARM64_IMAGE) --platform linux/arm64 --load .
 
-docker.build: clean $(BINARIES)
-	docker build -t $(IMAGE) --build-arg BUILD_FOLDER=. --build-arg TARGETPLATFORM= .
+.PHONY: docker.build.armv7
+docker.build.armv7: clean build.linux.armv7 docker.build.enable ## build linux/arm/v7 image
+	docker buildx build -t $(ARM_IMAGE) --platform linux/arm/v7 --load .
 
-docker-build: docker.build.amd64 docker.build.arm64 docker.build.armv7
-docker.build.amd64: clean build.linux.amd64 docker.build.enable
-	docker buildx build -t $(IMAGE) --platform linux/amd64 -f Dockerfile --load .
-docker.build.arm64: clean build.linux.arm64 docker.build.enable
-	docker buildx build -t $(ARM64_IMAGE) --platform linux/arm64 -f Dockerfile.arm64 --load .
-docker.build.armv7: clean build.linux.armv7 docker.build.enable
-	docker buildx build -t $(ARM_IMAGE) --platform linux/arm/v7 -f Dockerfile.armv7 --load .
+.PHONY: docker.push.local
+docker.push.local: ## push local image
+	docker push $(IMAGE)
 
-docker-push: docker.push.amd64 docker.push.arm64 docker.push.armv7
-docker.push.amd64:
+.PHONY: docker.push.amd64
+docker.push.amd64: ## push linux/amd64 image
 	docker push $(IMAGE)
-docker.push.arm64: docker.build.arm64
+
+.PHONY: docker.push.arm64
+docker.push.arm64: ## push linux/arm64 image
 	docker push $(ARM64_IMAGE)
-docker.push.armv7: docker.build.armv7
+
+.PHONY: docker.push.armv7
+docker.push.armv7: ## push linux/arm/v7 image
 	docker push $(ARM_IMAGE)
 
-# build multi-arch container image using a trusted multi-arch base image
-docker.push.multiarch: clean build.linux docker.build.enable
+# Currently it is not possible to build and push multiarch image separately,
+# see https://github.com/docker/buildx/issues/59
+.PHONY: docker.push.multiarch
+docker.push.multiarch: clean build.linux docker.build.enable ## build and push multi-arch image using a trusted multi-arch alpine base image
 	docker buildx build --rm -t $(MULTIARCH_IMAGE) --platform linux/amd64,linux/arm64 --push \
-	  --build-arg BASE_IMAGE=container-registry.zalando.net/library/alpine-3:latest .
+		--build-arg BASE_IMAGE=container-registry.zalando.net/library/alpine-3:latest .
```

Comment:
Similar to https://github.com/zalando/skipper/pull/2566/files#r1320057345. This is the right approach for targeting Zalando clusters but for the open source images (dockerhub?) `alpine:latest` would also work here.

Reply:
But like described above, it's perfectly fine to use a multi-arch base image (either `alpine:latest` or `container-registry.zalando.net/library/alpine-3:latest`) to build a single-arch image (i.e. `--platform linux/amd64` or `--platform linux/arm64`).

Reply:
I'm assuming you're using `docker buildx build`. For the places where you're still using `docker build` (if any, I didn't check) the outcome is different. Better to stick with `buildx` everywhere.